common/beerocks/bwl/common/base_wlan_hal.cpp

```diff
@@ -32,6 +33,9 @@ base_wlan_hal::base_wlan_hal(HALType type, std::string iface_name, IfaceType ifa
 
     // Initialize complex containers of the radio_info structure
     m_radio_info.supported_channels.resize(128 /* TODO: Get real value */);
+
+    m_nl80211_client = nl80211_client_factory::create_instance();
```

Comment:
nl80211_client is a dependency and thus should be injected, not hard-coded.
If we ever create a unit test for this class, nl80211_client should be provided as a parameter.

Reply:
I'm not sure I follow - can you explain what is hard coded here and what do you mean by injecting it?

Reply:
`nl80211_client` is a dependency for `base_wlan_hal`. `base_wlan_hal` should not create it. `nl80211_client` should be injected as a parameter to the constructor of `base_wlan_hal`.

But this is only actually required if we want to create a unit test for `base_wlan_hal`. In such unit test, we would pass a mock `nl80211_client` instead of the real implementation.

Reply:
Done